### PR TITLE
chore(checkbox): reduced specificity of scss

### PR DESCRIPTION
**Pull request by moog16**
related: #1687

example: 

WAS: 
`.mdc-checkbox .mdc-checkbox__native-control:disabled:not(:checked):not(:indeterminate) ~ .mdc-checkbox__background`

NOW:
`.mdc-checkbox__native-control:disabled:not(:checked):not(:indeterminate) ~ .mdc-checkbox__background`

### CODE REVIEW

File: packages/mdc-checkbox/_mixins.scss [hunk 1]
```diff
@@ -102,7 +102,9 @@
   white-space: nowrap;
   cursor: pointer;
   vertical-align: bottom;
+}
 
+@mixin mdc-checkbox-background-selector-disabled_ {
```
Comment (kfranqueiro) [inline packages/mdc-checkbox/_mixins.scss]: This doesn't seem like the right name for this mixin. Maybe dropping `selector-` is enough, but cc @advorak for his opinion.

Mixins with "selector" in the name are general-purpose mixins used to embed `@content` within a particular selector, which is not what this does.

Comment (moog16) [reply packages/mdc-checkbox/_mixins.scss]: I see...I was just copying the names of the two mixins included in this one. How about `mdc-checkbox-disabled-background-base-fill-color`?

Comment (acdvorak) [reply packages/mdc-checkbox/_mixins.scss]: `mdc-checkbox-disabled-color_($color)` seems a little closer to what we want.

Then, in `mdc-checkbox.scss`, you can pass `$mdc-checkbox-disabled-color` as the parameter value.

Comment (acdvorak) [reply packages/mdc-checkbox/_mixins.scss]: After talking to @kfranqueiro, you don't need a `$color` param - you can leave it as-is and just rename the mixin to `mdc-checkbox-disabled-container-color_`.

**Pull request merged**